tests/resolve-lock/resolve_lock.go

```diff
@@ -0,0 +1,496 @@
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	pd "github.com/pingcap/pd/v4/client"
+	"github.com/pingcap/tidb/store/tikv"
+	"github.com/pingcap/tidb/store/tikv/oracle"
+	"github.com/pingcap/tidb/store/tikv/tikvrpc"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tipocket/pkg/cluster/types"
+	"github.com/pingcap/tipocket/pkg/core"
+	"github.com/pingcap/tipocket/util"
+)
+
+// Config is for resolveLockClient
+type Config struct {
+	Concurrency             int
+	EnableGreenGC           bool
+	TableSize               int
+	LocksPerRegion          int
+	GenerateLockConcurrency int
+}
+
+// CaseCreator creates resolveLockClient
+type CaseCreator struct {
+	Cfg *Config
+}
+
+func (l CaseCreator) Create(node types.ClientNode) core.Client {
+	return &resolveLockClient{
+		Config: l.Cfg,
+	}
+}
+
+type resolveLockClient struct {
+	*Config
+	db *sql.DB
+
+	pd pd.Client
+	kv tikv.Storage
+}
+
+func (c *resolveLockClient) SetUp(ctx context.Context, nodes []types.ClientNode, idx int) error {
+	if idx != 0 {
+		return nil
+	}
+
+	var err error
+	node := nodes[idx]
+	dsn := fmt.Sprintf("root@tcp(%s:%d)/test", node.IP, node.Port)
+
+	log.Infof("start to init...")
+
+	c.db, err = util.OpenDB(dsn, c.Concurrency)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		log.Infof("init end...")
+	}()
+
+	// Disable GC
+	_, err = c.db.Exec(`update mysql.tidb set VARIABLE_VALUE = "10000h" where VARIABLE_NAME in ("tikv_gc_run_interval", "tikv_gc_life_time")`)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// We are not going to let TiDB manage the GC process. So this config is actually useless.
+	//scanLockMode := "legacy"
+	//if c.EnableGreenGC {
+	//	scanLockMode = "physical"
+	//}
+	//_, err = c.db.Exec(`insert into mysql.tidb values ('tikv_gc_scan_lock_mode', ?, '')
+	//		       on duplicate key
+	//		       update variable_value = ?, comment = ''`, scanLockMode, scanLockMode)
+	//if err != nil {
+	//	return errors.Trace(err)
+	//}
+
+	if c.TableSize == 0 {
+		log.Infof("table size is set to 0. Do not load data.")
+		return nil
+	}
+	_, err = c.db.Exec("create table t1(id integer, v varchar(128), primary key (id))")
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	const loadDataConcurrency = 100
+
+	errCh := make(chan error, loadDataConcurrency)
+	for threadNum := 0; threadNum < loadDataConcurrency; threadNum++ {
+		threadNum1 := threadNum
+		go func() {
+			for i := threadNum1; i < c.TableSize; i += loadDataConcurrency {
+				_, err := c.db.Exec("insert into t1 values(?, ?)", i, randStr())
+				if err != nil {
+					errCh <- errors.Trace(err)
+					return
+				}
+			}
+			errCh <- nil
+		}()
+	}
+
+	for i := 0; i < loadDataConcurrency; i++ {
+		err = <-errCh
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	// TODO: Is this the correct way to create the PD client and kv client?
```

Comment:
`nodes` only contain tidb pod, so you will get an empty set here now.

Reply:
maybe we need inject all nodes here. 😲

Reply:
Is there any way to access PD and TiKV?